rules/import-style.js

```diff
@@ -127,244 +128,382 @@ const defaultStyles = {
+								argumentsLength: 1,
+								optionalCall: false,
+								optionalMember: false,
+							})
+							&& (node.parent.type === 'ExpressionStatement' && node.parent.expression === node)
+						)) {
+							return;
+						}
+
+						const moduleName = getStringIfConstant(node.arguments[0], sourceCode.getScope(node.arguments[0]));
+						const allowedImportStyles = styles.get(moduleName);
+						const actualImportStyles = ['unassigned'];
+
+						report(node, moduleName, actualImportStyles, allowedImportStyles, true);
+					},
+					VariableDeclarator(node) {
+						if (!(
+							node.init?.type === 'CallExpression'
+							&& node.init.callee.type === 'Identifier'
+							&& node.init.callee.name === 'require'
+						)) {
+							return;
+						}
+
+						const assignmentTargetNode = node.id;
+						const moduleNameNode = node.init.arguments[0];
+						const moduleName = getStringIfConstant(moduleNameNode, sourceCode.getScope(moduleNameNode));
+
+						if (!moduleName) {
+							return;
+						}
+
+						const allowedImportStyles = styles.get(moduleName);
+						const actualImportStyles = getActualAssignmentTargetImportStyles(assignmentTargetNode);
+
+						report(node, moduleName, actualImportStyles, allowedImportStyles, true);
+					},
+				}
+				: {}),
+		};
+	},
+	meta: {
+		type: 'problem',
+		docs: {
+			description: 'Enforce specific import styles per module.',
+			recommended: true,
+		},
+		fixable: 'code',
+		schema: {
+			type: 'array',
+			additionalItems: false,
+			items: [
+				{
+					type: 'object',
+					additionalProperties: false,
+					properties: {
+						checkImport: {
+							type: 'boolean',
+						},
+						checkDynamicImport: {
+							type: 'boolean',
+						},
+						checkExportFrom: {
+							type: 'boolean',
+						},
+						checkRequire: {
+							type: 'boolean',
+						},
+						extendDefaultStyles: {
+							type: 'boolean',
+						},
+						styles: {
+							$ref: '#/definitions/moduleStyles',
+						},
+					},
 				},
-				extendDefaultStyles: {
-					type: 'boolean',
+			],
+			definitions: {
+				moduleStyles: {
+					type: 'object',
+					additionalProperties: {
+						$ref: '#/definitions/styles',
+					},
 				},
 				styles: {
-					$ref: '#/definitions/moduleStyles',
-				},
-			},
-		},
-	],
-	definitions: {
-		moduleStyles: {
-			type: 'object',
-			additionalProperties: {
-				$ref: '#/definitions/styles',
-			},
-		},
-		styles: {
-			anyOf: [
-				{
-					enum: [
-						false,
+					anyOf: [
+						{
+							enum: [
+								false,
+							],
+						},
+						{
+							$ref: '#/definitions/booleanObject',
+						},
 					],
 				},
-				{
-					$ref: '#/definitions/booleanObject',
+				booleanObject: {
+					type: 'object',
+					additionalProperties: {
+						type: 'boolean',
+					},
 				},
-			],
-		},
-		booleanObject: {
-			type: 'object',
-			additionalProperties: {
-				type: 'boolean',
 			},
 		},
-	},
-};
-
-/** @type {import('eslint').Rule.RuleModule} */
-const config = {
```

Comment:
The `create` method should not be inlined in the `config` object, for readability.

https://github.com/CasperEngl/eslint-plugin-unicorn/blob/2c6b0f7690e5a77d61f397d0d08b2cab7e6f690b/rules/catch-error-name.js#L124-L126